OpenRA.Game/Map/MapCache.cs

```diff
@@ -174,7 +174,8 @@ void LoadAsyncInternal()
 					if (bitmap == null)
 					{
 						createdPreview = true;
-						bitmap = Minimap.RenderMapPreview(modData.DefaultRules.TileSets[p.Map.Tileset], p.Map, modData.DefaultRules, true);
+						var map = new Map(p.Path);
```

Comment:
The Map constructor simply isn't thread-safe - so this change is bogus by default unless you fix that.

The fact this thread loads one at a time mostly saves you - but I can contrive some scenarios where it still explodes - such as switching mods whilst this thread is still generating previews and causing the global `Game.ModData` to change.

You could so something terribly hacky be reworking the Map constructor to do something like:
```
Map(string path)
{
    lock(StaticSyncObjectWhichIsLameButHeyItsBetterThanRandomThreadBugs)
    {
        // Load here
    }
}